### ISSUE TITLE
Current `aria-current` heuristics violate WAI-ARIA rules

### ISSUE THREAD

**Issue by binyamin** [solidjs/solid-router#289]
### Describe the bug

Whenever the `<A />` component is active, it adds `aria-current="page"` to the rendered element. Since this package defaults to only matching the beginning of the route, this violates [WAI-ARIA for `aria-current`](https://w3c.github.io/aria/#aria-current) in two ways:
1. There can only be one element marked as "current" within the set of elements. Currently, if I link to both `/` and `/docs`, they will both be marked as "current".
2. The `page` value is meant to indicate the _page_ that's currently displayed. Currently, if I'm viewing `/docs/introduction`, a link to `/docs` will be marked as "current".

**Note**: I don't have a stackblitz account, and I have plenty of GitHub repositories already. If you feel my description isn't clear, I can provide a gist.

### Your Example Website or App

See above

### Steps to Reproduce the Bug or Issue

See above

### Expected behavior

See above

### Platform

- OS: any
- Browser: any
- Version: `0.8.3`

**Comment by binyamin** [comment]
@ryansolid Do you need any more info?

**Comment by ryansolid** [comment]
I see that probably came from a PR. I haven't looked at this myself. Are you suggesting we just stop setting this property?

**Comment by binyamin** [comment]
Maybe. I see two options:
1. Apply `aria-current="page"` as if the `end` property is specified. Meaning, the `end` property will only affect the class-names.
2. Stop setting `aria-current`. As [they say](https://www.w3.org/WAI/ARIA/apg/practices/read-me-first/), "no ARIA is better than bad ARIA".

**Comment by ryansolid** [comment]
fix by #363
